serpentTools/parsers/microxs.py

```diff
@@ -0,0 +1,271 @@
+        Raises
+        ------
+        SerpentToolsException:
+            If energy is a negative number
+            If parent or fission product are not found
+        """
+        IndYield, CumYield = 0, 0
+        if energy < 0:
+            raise SerpentToolsException("Energy entry {0} must be positive"
+                                        " in {1}".format(energy, self.filePath))
+        eneList = []
+        if flagEnergy:
+            # obtain the different energies for a specific parent
+            eneList = [items[1] for items in self.nfy.keys() if parent == items[0]]
+            # obtain the closest energy value
+            energy = min(eneList,
+                         key=lambda x: abs(x - energy)) if eneList else energy
+        if (parent, energy) not in self.nfy.keys():
+            raise SerpentToolsException("There is no parent {0} with energy "
+                                        "{1} in {2}".format(parent,energy,
+                                                            self.filePath))
+
+        FP = self.nfy[(parent, energy)]['fissProd']
+        IndYield = self.nfy[(parent, energy)]['indYield']
+        CumYield = self.nfy[(parent, energy)]['cumYield']
+        if daughter:
+            IndYield = IndYield[FP == daughter]
+            CumYield = CumYield[FP == daughter]
+        return IndYield, CumYield
+
+
+    def getXS(self, universe, isotope, reaction, isomeric=0):
+        """
+        Return the group-wise micro cross-sections for a
+        specific isotope, and reaction
+
+        Parameters
+        ----------
+        universe: string
+            universe ID, e.g., `0`
+        isotope: int or float
+            ID of the isotope (ZZAAA0/1)
+        reaction: int
+            MT reaction, e.g., 102 --> (n,gamma)
+        Special flag: int or float
+            Isomeric state or fission yield distribution number
+            Default is zero
+
+        Returns
+        -------
+        Group-wise micro cross-sections: list
+            if the isotope, reaction or isomeric state
+            do not exist, an empty list is returned
+
+        Raises
+        ------
+        SerpentToolsException:
+            If the universe does not exist
+            If the isotope's format is incorrect (not ZZAAA0/1)
+        """
+        if universe not in self.xsVal.keys():
+            raise SerpentToolsException("Universe {0} does not exist in {1}"
+                                        .format(universe, self.filePath))
+        if isotope < 10010 or isotope > 1000000:
+            raise SerpentToolsException("Isotope {0} is not properly formatted"
+                                        " ZZAAA0/1 in {1}"
+                                        .format(isotope, self.filePath))
+        if (isotope, reaction, isomeric) in self.xsVal[universe]:
+            return self.xsVal[universe][(isotope, reaction, isomeric)], \
+                   self.xsUnc[universe][(isotope, reaction, isomeric)]
+        else:
```

Comment:
`else` statement here is not needed, as the above `if` returns.

Reply:
Modified.